components/match2/wikis/clashroyale/match_group_input_custom.lua

```diff
@@ -0,0 +1,725 @@
+
+function walkoverProcessing.walkover(obj, scores)
+	local walkover = obj.walkover
+
+	if Logic.isNumeric(walkover) then
+		walkoverProcessing.numericWalkover(obj, walkover)
+	elseif walkover then
+		walkoverProcessing.nonNumericWalkover(obj, walkover)
+	elseif #scores ~=2 then -- since we always have 2 opponents
+		error('Unexpected number of opponents when calculating winner')
+	elseif Array.all(scores, function(score)
+			return Table.includes(ALLOWED_STATUSES, score) and score ~= DEFAULT_WIN_STATUS
+		end) then
+
+		walkoverProcessing.scoreDoubleWalkover(obj, scores)
+	elseif Array.any(scores, function(score) return Table.includes(ALLOWED_STATUSES, score) end) then
+		walkoverProcessing.scoreWalkover(obj, scores)
+	end
+end
+
+function walkoverProcessing.numericWalkover(obj, walkover)
+	local winner = tonumber(walkover)
+
+	obj.winner = winner
+	obj.finished = true
+	obj.walkover = UNKNOWN_REASON_LOSS_STATUS
+	obj.resulttype = DEFAULT_WIN_RESULTTYPE
+end
+
+function walkoverProcessing.nonNumericWalkover(obj, walkover)
+	if not Table.includes(ALLOWED_STATUSES, string.upper(walkover)) then
+		error('Invalid walkover "' .. walkover .. '"')
+	elseif not Logic.isNumeric(obj.winner) then
+		error('Walkover without winner specified')
+	end
+
+	obj.winner = tonumber(obj.winner)
+	obj.finished = true
+	obj.walkover = walkover:upper()
+	obj.resulttype = DEFAULT_WIN_RESULTTYPE
+end
+
+function walkoverProcessing.scoreDoubleWalkover(obj, scores)
+	obj.winner = -1
+	obj.finished = true
+	obj.walkover = scores[1]
+	obj.resulttype = DEFAULT_WIN_RESULTTYPE
+end
+
+function walkoverProcessing.scoreWalkover(obj, scores)
+	local winner, status
+
+	for scoreIndex, score in pairs(scores) do
+		score = string.upper(score)
+		if score == DEFAULT_WIN_STATUS then
+			winner = scoreIndex
+		elseif Table.includes(ALLOWED_STATUSES, score) then
+			status = score
+		else
+			status = UNKNOWN_REASON_LOSS_STATUS
+		end
+	end
+
+	if not winner then
+		error('Invalid score combination "{' .. scores[1] .. ', ' .. scores[2] .. '}"')
+	end
+
+	obj.winner = winner
+	obj.finished = true
+	obj.walkover = status
+	obj.resulttype = DEFAULT_WIN_RESULTTYPE
+end
+
+function walkoverProcessing.applyMatchWalkoverToOpponents(match)
+	for opponentIndex = 1, MAX_NUM_OPPONENTS do
+		local score = match['opponent' .. opponentIndex].score
+
+		if Logic.isNumeric(score) or String.isEmpty(score) then
+			match['opponent' .. opponentIndex].score = String.isNotEmpty(score) and score or NO_SCORE
+			match['opponent' .. opponentIndex].status = match.walkover
+		elseif score and Table.includes(ALLOWED_STATUSES, score:upper()) then
+			match['opponent' .. opponentIndex].score = NO_SCORE
+			match['opponent' .. opponentIndex].status = score
+		else
+			error('Invalid score "' .. score .. '"')
+		end
+	end
+
+	-- neither match.opponent0 nor match['opponent-1'] does exist hence the if
+	if match['opponent' .. match.winner] then
+		match['opponent' .. match.winner].status = DEFAULT_WIN_STATUS
+	end
+end
```

Comment:
check what of this can be replaced with commons stuff

Reply:
unless im blind (which is very possible), Commons doesnt have any methods handling walkovers. Only Forfeits